src/pages/index.js

```diff
@@ -29,54 +31,195 @@ const enableValidation = (config) => {
+        api
+          .disLikeCard(id)
+          .then(() => card._handleLikeIcon())
+          .catch((err) => console.error(`${err}, Failed to dislike card`));
+      }
+    }
+  );
   return card.getView();
 };
 
+//Popups
+
 const imgPopup = new PopupWithImages("#card-picture-modal");
 imgPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
-  },
-  cardListElement
+const profilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileUpdate
 );
-cardSection.renderItems();
-
-const userInfo = new UserInfo(".profile__name", ".profile__job");
-
-const profilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
-  profilePopup.close();
-  userInfo.setUserInfo(data);
-});
 profilePopup.setEventListeners();
 
-const newCardPopup = new PopupWithForm("#card-add-modal", (data) => {
-  console.log(data);
-  const cardFormElement = createCard(data);
-  cardSection.addItem(cardFormElement);
-  newCardPopup.close();
-});
+const newCardPopup = new PopupWithForm("#card-add-modal", handleAddCard);
 newCardPopup.setEventListeners();
 
+export const deleteConfirmPopup = new PopupConfirmation(
+  "#modal-delete",
+  handleDeleteConfirmation
+);
+deleteConfirmPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#modal-avatar", handleUpdateAvatar);
+avatarPopup.setEventListeners();
+
+//HANDLES
+
+// function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+//   popupInstance.renderLoading(true, loadingText);
+//   request()
+//     .then(() => {
+//       popupInstance.close();
+//     })
+//     .catch(console.error)
+//     .finally(() => {
+//       popupInstance.renderLoading(false);
+//     });
+// }
+
+// function handleProfileFormSubmit(inputValues) {
+//         we create a function that returns a promise
+//   function makeRequest() {
+//        `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+//     return api.editProfile(inputValues).then((userData) => {
+//       userInfo.setUserInfo(userData)
+//     });
+//   }
+//         Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
+//   handleSubmit(makeRequest, profilePopup);
+// }
+
+function handleDeleteConfirmation(card, id) {
+  api
+    .deleteCard(id)
+    .then(() => {
+      card.handleDeleteCard();
+      deleteConfirmPopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to delete card`));
+}
+
+function handleProfileUpdate(userData) {
+  profilePopup.setLoading(true);
+  api
+    .updateUserData(userData)
+    .then(() => {
+      initialUserData.setUserInfo(userData);
+      profilePopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to update Profile Info`))
+    .finally(() => {
+      profilePopup.setLoading(false);
```

Comment:
`.setLoading(false)` is not needed in `finally`, because you have `setSave` for that

Reply:
I went with your initial suggestion I missed earlier, so now its the renderLoading